### PR TITLE
[1.20] Automated cherry pick of #98477: update gogo/protobuf to v1.3.2

### PR DESCRIPTION
**Pull request by joelsmith**
Cherry pick of #98477 on release-1.20.

#98477: update gogo/protobuf to v1.3.2

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note
Regenerate protobuf code to fix CVE-2021-3121
```

### CODE REVIEW

**Comment by joelsmith** [comment]
@kubernetes/dep-approvers I would appreciate a review. We need to bump gogo/protobuf to v1.3.2, but doing so updates other deps too. Is that okay on a prior version?

**Comment by sttts** [comment]
automated cherry-pick with generated code are suspicious. Have you checked that updating the proto files afterwards is a noop?

**Comment by sttts** [comment]
/lgtm
/approve

for dep changes.

/assign @liggitt 

for api changes.

**Comment by joelsmith** [comment]
> automated cherry-pick with generated code are suspicious. Have you checked that updating the proto files afterwards is a noop?

Although this PR started as an automated cherry pick, I had to re-run several of the generation steps until all of the code-generation verification checks were happy.

```
hack/pin-dependency.sh github.com/gogo/protobuf v1.3.2
hack/update-vendor.sh
hack/lint-dependencies.sh
# and repeat for any identified additional deps

hack/update-generated-kubelet-plugin-registration.sh
hack/update-generated-pod-resources.sh
hack/update-generated-protobuf.sh
```

File: go.mod [hunk 1]
```diff
@@ -125,7 +125,7 @@ require (
 	k8s.io/controller-manager v0.0.0
 	k8s.io/cri-api v0.0.0
 	k8s.io/csi-translation-lib v0.0.0
-	k8s.io/gengo v0.0.0-20201113003025-83324d819ded
+	k8s.io/gengo v0.0.0-20201214224949-b6c5ce23f027
```
Comment (liggitt) [inline go.mod:128]: why was this needed? this wasn't in the `master` PR, right?

File: pkg/apis/apps/v1/zz_generated.defaults.go [hunk 2]
```diff
@@ -109,7 +107,7 @@ func SetObjectDefaults_DaemonSet(in *v1.DaemonSet) {
 		corev1.SetDefaults_Container(a)
 		for j := range a.Ports {
 			b := &a.Ports[j]
-			if reflect.ValueOf(b.Protocol).IsZero() {
```
Comment (liggitt) [inline pkg/apis/apps/v1/zz_generated.defaults.go:112]: the `zz_generated.defaults.go` changes are unexpected and were not in the master PR

**Comment by joelsmith** [comment]
@liggitt thanks for catching the extra chages. I think that happened when I tried to use the automatic cherry-pick process with manual conflict resolution, then did auto-generation from there. I probably made mistakes along the way. I started over from release-1.20, and that got rid of the extra changes.

File: pkg/kubelet/patch [hunk 3]
```diff
@@ -0,0 +1,102 @@
+diff --git a/pkg/kubelet/kubelet.go b/pkg/kubelet/kubelet.go
```
Comment (liggitt) [inline pkg/kubelet/patch:1]: drop this?

Comment (joelsmith) [reply pkg/kubelet/patch]: Thanks, dunno how I git-added that. Removed.

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**